[ads] Only collect, store and redeem essential metadata (#24188)
brave/brave-core: components/brave_ads/core/internal/account/user_data/README.md

| createdAtTimestamp  | {"createdAtTimestamp":"2020-11-18T12:00:00.000Z"}  | [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp with fixed values for minutes, seconds, milliseconds and timezone.  |
| diagnosticId[^1][^2]  | {"diagnosticId":"c1298fde-7fdb-401f-a3ce-0b58fe86e6e2"}  | Diagnostic id from [brave://rewards-internals](brave://rewards-internals) to assist in troubleshooting issues. This is only included if it has been set manually by the user, for example, at the request of Brave Support in order to troubleshoot a problem.  |
| httpResponseStatus[^2]  | {"httpResponseStatus":"errorPage"}  | Indicates whether landing on a page resulted in an error page. This is only included if an error occurred.  |
| platform  | {"platform":"windows"}  | Operating system.  |

> | rotatingHash[^2]  | {"rotatingHash":"j9D7eKSoPLYNfxkG2Mx+SbgKJ9hcKg1QwDB8B5qxlpk="}  | A Base64-encoded, privacy-preserving, time-based hash value that integrates time as an essential element in its computation, ensuring the hash's validity is confined to a designated period.  |

Reviewer: ```suggestion
| rotatingHash[^2]  | {"rotatingHash":"j9D7eKSoPLYNfxkG2Mx+SbgKJ9hcKg1QwDB8B5qxlpk="}  | Hashed device identifier that is unique to each ad and rotated several times a day. This is used for rate-limiting purposes. |
```

# User Data

Return to [confirmation redemption](../../account/utility/redeem_confirmation/README.md).

> [!IMPORTANT]
> Approval from the privacy team is required for changes to existing user data or for new user data.

## Non-Brave Rewards User

### User Data Submitted when Redeeming Anonymous Confirmations

Included as part of the "confirmation" payload. See [non-Brave Rewards user confirmation payload](../confirmations/non_reward/README.md).

| key  | example  | description  |
|---|---|---|

> | conversion.action[^2]  | {"conversion":[{"action":"click"}]}  | Click-through conversion attribution to determine which advertising efforts led a user to click on an ad and subsequently complete a desired action.  |

Reviewer: Click-through conversions is the only possibility for Search Ads, but there's more to include here for Brave News, right?

Author: Great catch. I will update. Thank you.

Author: Added. Thanks

> | conversion.action[^2]  | {"conversion":[{"action":"click"}]}  | Conversion attribution to determine which advertising efforts led a user to view or click on an ad and subsequently complete a desired action.<br><br> Supported types:<br><br>Brave Search ads:<br>- click-through<br><br>Brave News ads:<br>- view-through<br>- click-through  |

Reviewer: For the _Supported types_, you could also link to the table at https://ads-help.brave.com/campaign-performance/reporting/#available-reporting-metrics-in-brave-ads-manager to avoid having another copy of the same information.

Author: Added. Thanks

## Brave Rewards User

### User Data Submitted when Redeeming Anonymous Confirmation Tokens

Included as part of the "confirmation token" payload. See [Brave Rewards user confirmation token payload](../confirmations/reward/README.md).

| key  | example  | description  |
|---|---|---|
| buildChannel  | {"buildChannel": "beta"}  | Browser build channel.  |

> | catalog  | {"catalog":[{"id":"29e5c8bc0ba319069980bb390d8e8f9b58c05a20"}]}  |  Catalog identifier.  |

Reviewer: ```suggestion
| catalog  | {"catalog":[{"id":"29e5c8bc0ba319069980bb390d8e8f9b58c05a20"}]}  |  Refers to the catalog that the ad was pulled from. |
```

| conversion.action[^2]  | {"conversion":[{"action":"view"}]}  | View-through or click-through conversion attribution to determine which advertising efforts led a user to view or click on an ad and subsequently complete a desired action.  |

> | conversion.envelope[^2]  | {"conversion":[{"envelope":<br>{"alg":"crypto_box_curve25519xsalsa20poly1305",<br>"ciphertext":"3f8a8aeb5e1e4b1a8e7b4e3f8a8aeb5e1e4b1a8e7b4e3f8a8aeb5e1e4b1a8e7b",<br>"epk":"d1f2e3d4c5b6a79887a6b5c4d3e2f1d1f2e3d4c5b6a79887a6",<br>"nonce":"a1b2c3d4e5f6a7b8c9d0e1f2a3b4c5d6"}}]}  | Allow advertisers to affirm that conversions are legitimate by wrapping an identifier with an additional layer of encryption to protect its integrity.  |

Reviewer: Have we published how Verifiable Conversions work anywhere? Can we link to that?

Author: Added. Thanks

> | createdAtTimestamp  | {"createdAtTimestamp":"2020-11-18T12:00:00.000Z"}  | Privacy-preserving [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp that conceals minutes, seconds, and milliseconds.  |

Reviewer: ```suggestion
| createdAtTimestamp  | {"createdAtTimestamp":"2020-11-18T12:00:00.000Z"}  | [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp with fixed values for minutes, seconds, milliseconds and timezone. |
```

> | diagnosticId[^1][^2]  | {"diagnosticId":"c1298fde-7fdb-401f-a3ce-0b58fe86e6e2"}  | Diagnostic id from [brave://rewards-internals](brave://rewards-internals) to assist in troubleshooting issues. It is initially empty.  |

Reviewer: Is this right, or would there be another case in which there is a value here?
```suggestion
| diagnosticId[^1][^2]  | {"diagnosticId":"c1298fde-7fdb-401f-a3ce-0b58fe86e6e2"}  | Diagnostic id from [brave://rewards-internals](brave://rewards-internals) to assist in troubleshooting issues. This is empty unless it has been set manually by the user, for example at the request of Brave Support in order to troublehoot a problem. |
```

> | httpResponseStatus[^2]  | {"httpResponseStatus":"errorPage"}  | Whether a page land resulted in an error page.  |

Reviewer: ```suggestion
| httpResponseStatus[^2]  | {"httpResponseStatus":"errorPage"}  | A standard HTTP error code (e.g. 404) indicating whether a visit to the landing page resulted in an error page. |
```

Reviewer: What do we send in the case of non-errors?

Author: If an error occurred we return "errorPage" and do not return the actual error at this time.

Reviewer: So it's either:
```
{"httpResponseStatus":"404"}
```
(or another status code number), or:
```
{"httpResponseStatus":"errorPage"} 
```
?

Author: If an error occurred the payload will only ever include `{"httpResponseStatus":"errorPage"} `

Reviewer: Ah sorry, I misread your comment.

And what do we return if there's no error?
```
{}
```
?

Author: I have updated the comment :-). Thanks

> | locale[^2]  | {"countryCode":"US"}  | Privacy-preserving operating system locale. See [k-Anonymity](../../common/locale/country_code_anonymity_util.cc).  |

Reviewer: ```suggestion
| locale[^2]  | {"countryCode":"US"}  | Country code reported by the operating system. Small countries (based on population) are not reported. See [k-Anonymity](../../common/locale/country_code_anonymity_util.cc) for the list. |
```

Reviewer: What do we report instead for small countries?

| platform  | {"platform":"windows"}  | Operating system.  |
| rotatingHash[^2]  | {"rotatingHash":"j9D7eKSoPLYNfxkG2Mx+SbgKJ9hcKg1QwDB8B5qxlpk="}  | A Base64-encoded, privacy-preserving, time-based hash value that integrates time as an essential element in its computation, ensuring the hash's validity is confined to a designated period.  |
| segment[^2]  | {"segment":"sports"}  | Advertising taxonomy for the chosen ad.  |
| studies  | {"studies":[{"group":"GroupA","name":"BraveAds.FooStudy"},{"group":"GroupB","name":"BraveAds.BarStudy"}]}  | User studies, utilized for A/B testing, should not exceed one active study at any time. Configuration of studies is achievable through Griffin by adding "BraveAds." as a prefix to the experiment name. See [Griffin](https://github.com/brave/brave-browser/wiki/Brave-Variations-(Griffin)).  |

> | systemTimestamp[^1]  | {"systemTimestamp": "2020-11-18T12:00:00.000Z"}  | Privacy-preserving [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp that conceals minutes, seconds, and milliseconds.  |

Reviewer: Like above:
```suggestion
| systemTimestamp[^1]  | {"systemTimestamp": "2020-11-18T12:00:00.000Z"}  | [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp with fixed values for minutes, seconds, milliseconds and timezone.  |
```
